### PR TITLE
Remove unnecessary deps and apt-get upgrade

### PR DESCRIPTION
**Pull request by jgmize**
@glogiotatidis r?

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,8 @@
 FROM python:3.4-wheezy
-RUN apt-get update && apt-get install -y libpq-dev gettext libjpeg8-dev postgresql-client
+RUN apt-get update &&\
+    apt-get install -y libpq-dev gettext libjpeg8-dev postgresql-client &&\
```
Comment (glogiotatidis) [inline Dockerfile]: Let's change this to `apt-get install -y --no-install-recommends libpq-dev gettext libjpeg8-dev postgresql-client` and it won't install recommended packages and among them is python-minimal.

File: Dockerfile [hunk 2]
```diff
@@ -1,5 +1,8 @@
...
+    apt-get remove python mysql-common -y && apt-get autoremove -y &&\
```
Comment (glogiotatidis) [inline Dockerfile]: If we make the above change you don't need to remove python here. If we don't you want to remove `python-minimal` b/c removing `python` won't uninstall `python-minimal` neither `python2.7-minimal`

**Comment by glogiotatidis** [comment]
Docker's python image is bloated but that shouldn't hurt besides disk space. Maybe we can contribute upstream when we're done here.

**Comment by glogiotatidis** [comment]
`no-install-recommmends` is the way to go i believe. Other than that looks good r+wc :feet:

File: Dockerfile (hunk 1 above)
Comment (jgmize) [inline Dockerfile]: good idea, I'll update.

**Comment by jgmize** [comment]
@glogiotatidis thanks, I made your suggested changes.

**Pull request merged**